pyanaconda/payload/requirement.py

```diff
@@ -118,6 +118,20 @@ def add_groups(self, group_ids, reason, strong=True):
         """
         self._add(PayloadRequirementType.group, group_ids, reason, strong)
 
+    def add_requirements(self, requirements):
+        """Add requirements from a list of Requirement instances.
+
+        :param requirements: list of Requirement instances
+        """
+        for requirement in requirements:
+            # check requirement type and add a payload requirement appropriately
+            if requirement.type == "package":
+                self.add_packages([requirement.name], reason=requirement.reason)
+            elif requirement.type == "group":
+                self.add_groups([requirement.name], reason=requirement.reason)
```

Comment:
You can use the enum `PayloadRequirementType` and the method `_add` to simplify this code if you want to. However, this implementation is good enough for now.

Reply:
I think its fine for now. :)